### PR TITLE
add lambda-cognito webhook example (fix #1180)

### LINKED ISSUE #1180: Example for using auth with AWS Cognito
Document/illustrate https://aws.amazon.com/cognito/ as auth webhook.

### PR DESCRIPTION
**Pull request by Nomad-Go**
### Description

Adding a webhook example for Cognito in a Lambda function.

What component does this PR affect? 

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [X ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue

[1180](https://github.com/hasura/graphql-engine/issues/1180)

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X ] Docs update
- [ X] Community content

### Checklist:

- [X ] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ X] This change requires a change in the documentation. 
- [ X] I have updated the documentation accordingly.
- [ X] I have added required tests.

### CODE REVIEW

**Comment by shahidhk** [comment]
@Nomad-Go Can you sign the CLA so that we can get this merged?

**Comment by Nomad-Go** [comment]
Roger. It's telling me I have signed it now when I click on it.

**Comment by ecthiender** [comment]
@Nomad-Go the author that made those commits do not have the same email address as your account. Hence it still reports that CLA is not signed by the committer (even your account has signed the CLA).

File: community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    /*
+    This is all from the sample AWS Code layed out in the first link above. The only difference is that I made an async/ await compatable get request for the token. I also provide 
+    but commented out a modified sample to compare the audience key in the token to multiple application client secrets. Application client secrets are made from the cognito console. 
+    The thoght is that there could be multiple client, each with their own keys, using Hasura.
+    */
+
+    //1) Parse out and decode the Key ID (kid) from the incomming token.
+    const sections = token.split(".");
+    var header = jose.util.base64url.decode(sections[0]);
+    header = JSON.parse(header);
```
Comment (ecthiender) [inline community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js]: @Nomad-Go I'm not sure if you have to manually parse the parts of the token? Doesn't the `jose` library have APIs which let's you get the `kid` directly ?

**Comment by marionschleifer** [comment]
@Nomad-Go we would love to merge this PR 🙂 could you please address Anon's comment in the last review?

File: community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js (hunk 1 above)
Comment (Nomad-Go) [reply community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js]: Let me change this code to see if I can get the KID using the `jose` library built in functions and test it in my companies development environment. I looks like AWS uses the `RS256` algorithm as described [here](https://aws.amazon.com/premiumsupport/knowledge-center/decode-verify-cognito-json-token/).

File: community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    const kid = header.kid;
```
Comment (Nomad-Go) [inline community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js]:
```suggestion
    const pieces = jose.parse(token);
    const kid = pieces.header.kid;
```

File: community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js (hunk 1 above)
Comment (Nomad-Go) [reply community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js]: I left another comment with a change to parse the kid using the `jose` library.

**Pull request merged**